### ISSUE TITLE
GridTools C++ v2 version should be pinned

### ISSUE THREAD

**Issue by jdahm** [GridTools/gt4py#457]
The GridTools master branch is checked out instead of a pinned version for the GTC backends, so updates in that project could break gt4py. We should pin this to a release branch or specific version.

In addition, the `gt_src_manager` simply clones the GridTools repository branches, but should probably strip docs and examples, to reduce the number of files.

**Comment by jdahm** [comment]
@havogt is there something we are waiting for in order to tag the version in the gridtools/Gridtools repo?

**Comment by jdahm** [comment]
@havogt is there something we are waiting for in order to tag the version in the gridtools/Gridtools repo?

**Comment by havogt** [comment]
There is no blocker. I will do it after my vacation.

**Comment by jdahm** [comment]
Related issue that might require another issue here: the regression tests in this repository use non-installed headers from GridTools. So when I install GridTools and GridTools v2 using this nix derivations https://github.com/VulcanClimateModeling/packages/tree/master/gridtools, and setting `GT_INCLUDE_DIR` and `GT2_INCLUDE_DIR`, then I get the following error from pytest:
```
/Users/johannd/Code/gt4py/tests/reference_cpp_regression/reference.cpp:17:10: fatal error: 'gridtools/regression/horizontal_diffusion_repository.hpp' file not found
#include "gridtools/regression/horizontal_diffusion_repository.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

For now I will change the derivations to manually copy over the regression directory to the installation path, but it's a workaround.

**Comment by havogt** [comment]
I see 2 solutions:
- In GT4Py, remove the dependency on GridTools C++ test header. I don't know if this test is valuable.
- Declare that GT4Py ships GridTools C++, then there is no problem. I would argue that this is currently the only supported way.

I don't fully understand what you are trying to do in https://github.com/VulcanClimateModeling/packages/tree/master/gridtools/v1.nix (and additionally the link is private).

**Comment by jdahm** [comment]
@havogt Just a ping again that this would be great to have.

**Comment by havogt** [comment]
I moved the problem of the test header into a separate issue. Let's focus here on the discussion of the tag.

**Comment by havogt** [comment]
What's the concrete problem you want to solve?

Actually I think using master is the right strategy for GT4Py as GT4Py is the main driver of changes in GridTools and both are maintained by the same team. I see it as the same package.

**Comment by jdahm** [comment]
I get that, but why not use PR mechanism to update the GridTools version in gt4py? That way we get reproducible results with a gt4py version.

**Comment by havogt** [comment]
> What's the concrete problem you want to solve?

My question was more about, did you run into a problem in the past which would be solved by this approach? I see more a problem of an old local GridTools clone not being updated when it should be, but that wouldn't be solved by tagging (unless we do checking of the version on each run).

**Comment by jdahm** [comment]
> did you run into a problem in the past which would be solved by this approach

To be fair, this has only once been an issue once. There was a breaking change introduced both into gt4py and gridtools, and since locally we had gridtools already downloaded via `gt_src_manager`, it was not updated, so after merging `master` into various feature branches we got strange errors. It took a moment for us to realize we should wipe the gridtools clone and let it re-download the latest version. If there were some mechanism to resolve that type of issue automatically that would be fantastic.

**Comment by havogt** [comment]
I think we discussed in some other place that fixing the version in the gt_src_manager doesn't solve the issue that you describe. I have no good idea to solve that:
- we can do a version check on each execution of a stencil but this introduces a penalty
- we can introduce a git hook that does a version check on pull, checkout, etc
- we can pin each commit to the latest GridTools version via submodules.

**Comment by jdahm** [comment]
I believe this is done now.
